criu/img-remote.c

```diff
@@ -1,26 +1,12 @@
-#include <unistd.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <sys/types.h>
-#include <sys/socket.h>
```

Comment:
In this file, sys/socket.h is included here because socket() is called in this file.

Reply:
We should not rely on criu headers in such cases. Maybe we should not include linux headers in criu headers...

Reply:
I agree that headers such as `sys/socket.h` and `sys/epoll.h` are used in this file and it is good to have them listed at the beginning as a hint about what functionalities are being used in the implementation that follows. However, if we were to do that consistently we will end up with a lot of duplicated code between *.c files and it is likely (in the future) for a developer to forget to remove an include statement when they remove a specific function call.
@avagin Considering both pros and cons, which headers do you think should be kept?
(I am happy to update the patch.)

Reply:
> However, if we were to do that consistently we will end up with a lot of duplicated code between *.c files and it is likely (in the future) for a developer to forget to remove an include statement when they remove a specific function call.

Do you know where sys/socket.h is included now? Is it a side effect? Does the same side effect exists in other libc-s?